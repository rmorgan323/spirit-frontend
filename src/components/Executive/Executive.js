import React from 'react';
import SessionHeader from '../SessionHeader/SessionHeader';
import CategorySliders from '../CategorySliders/CategorySliders';
import { executive } from '../../language/language';
import './Executive.css';

const Executive = () => {
  const { header, columns } = executive;
  return (
    <div className="Executive">
      <SessionHeader title={header.title} headings={[]} />
      <div className="category-content">
        {Object.keys(columns).map(column => {
          const { terms, db } = columns[column];

          return (
            <CategorySliders key={db} slidersArray={terms} databaseNamesArray={db} />
          );
        })}
      </div>
    </div>
  );
};

export default Executive;
