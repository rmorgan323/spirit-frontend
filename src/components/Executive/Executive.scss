.Executive {
  .category-content {
    display: flex;
    margin: auto;
    width: fit-content;
  }
}