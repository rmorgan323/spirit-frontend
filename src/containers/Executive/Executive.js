import React from 'react';
import SessionHeader from '../SessionHeader/SessionHeader';
import CategorySliders from '../../components/CategorySliders/CategorySliders';
import './Executive.css';

const Executive = () => {
  
  return (
    <div className="Executive">
      <SessionHeader 
        title={'Executive Functioning Capacities'}
        headings={[]}
      />
      <div className="category-content">
        <CategorySliders 
          slidersArray={['Orient EYES-EARS-SELF to GOAL', 'Inhibition']}
          databaseNamesArray={['exe_1_orient', 'exe_1_inhibition']}
        />
        <CategorySliders 
          slidersArray={['Working Memory', 'Shifting']}
          databaseNamesArray={['exe_2a_working', 'exe_2a_shifting']}
        />
        <CategorySliders 
          slidersArray={['Verbal', 'Spacial', 'General', 'Global/Local', 'Flexibility']}
          databaseNamesArray={['exe_2b_verbal', 'exe_2b_spatial', 'exe_2b_general', 'exe_2b_global', 'exe_2b_flexibility']}
        />
        <CategorySliders 
          slidersArray={['Inhibition', 'Shifting', 'Sequencing', 'Monitoring']}
          databaseNamesArray={['exe_3_inhibition', 'exe_3_shifting', 'exe_3_sequencing', 'exe_3_monitoring']}
        />
        <CategorySliders 
          slidersArray={['Future-Oriented/Goal-Directed', 'Spatial/Temporal Organization', 'Problem Solving', 'Reflective Thinking', 'Internalized Language']}
          databaseNamesArray={['exe_4a_future', 'exe_4a_spatial', 'exe_4a_problem', 'exe_4a_reflective', 'exe_4a_internalized']}
        />
        <CategorySliders 
          slidersArray={['Motivational Bias', 'Effortful Control', 'Self-Control', 'Self-Regulation', 'Self-Compassion']}
          databaseNamesArray={['exe_4b_motivational', 'exe_4b_effortful', 'exe_4b_self_control', 'exe_4b_self_regulation', 'exe_4b_self_compassion']}
        />
      </div>
    </div>
  )
}

export default Executive;