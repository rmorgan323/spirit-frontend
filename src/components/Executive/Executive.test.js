import React from 'react';
import Executive from './Executive';
import { shallow } from 'enzyme';

describe('Executive tests', () => {
  it('Should match snapshot', () => {
    const renderedApp = shallow(<Executive />);

    expect(renderedApp).toMatchSnapshot();
  });
});
