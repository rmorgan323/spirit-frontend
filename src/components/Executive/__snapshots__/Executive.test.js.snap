// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Executive tests Should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Executive />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SessionHeader
          headings={Array []}
          title="Executive Functioning Capacities"
        />,
        <div
          className="category-content"
        >
          <CategorySliders
            databaseNamesArray={
              Array [
                "exe_1_orient",
                "exe_1_inhibition",
              ]
            }
            slidersArray={
              Array [
                "Orient EYES-EARS-SELF to GOAL",
                "Inhibition",
              ]
            }
          />
          <CategorySliders
            databaseNamesArray={
              Array [
                "exe_2a_working",
                "exe_2a_shifting",
              ]
            }
            slidersArray={
              Array [
                "Working Memory",
                "Shifting",
              ]
            }
          />
          <CategorySliders
            databaseNamesArray={
              Array [
                "exe_2b_verbal",
                "exe_2b_spatial",
                "exe_2b_general",
                "exe_2b_global",
                "exe_2b_flexibility",
              ]
            }
            slidersArray={
              Array [
                "Verbal",
                "Spatial",
                "General",
                "Global/Local",
                "Flexibility",
              ]
            }
          />
          <CategorySliders
            databaseNamesArray={
              Array [
                "exe_3_initiation",
                "exe_3_planning",
                "exe_3_sequencing",
                "exe_3_monitoring",
              ]
            }
            slidersArray={
              Array [
                "Initiation",
                "Planning",
                "Sequencing",
                "Monitoring",
              ]
            }
          />
          <CategorySliders
            databaseNamesArray={
              Array [
                "exe_4a_future",
                "exe_4a_spatial",
                "exe_4a_problem",
                "exe_4a_reflective",
                "exe_4a_internalized",
              ]
            }
            slidersArray={
              Array [
                "Future-Oriented/Goal-Directed",
                "Spatial/Temporal Organization",
                "Problem Solving",
                "Reflective Thinking",
                "Internalized Language",
              ]
            }
          />
          <CategorySliders
            databaseNamesArray={
              Array [
                "exe_4b_motivational",
                "exe_4b_effortful",
                "exe_4b_self_control",
                "exe_4b_self_regulation",
                "exe_4b_self_compassion",
              ]
            }
            slidersArray={
              Array [
                "Motivational Bias",
                "Effortful Control",
                "Self-Control",
                "Self-Regulation",
                "Self-Compassion",
              ]
            }
          />
        </div>,
      ],
      "className": "Executive",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "headings": Array [],
          "title": "Executive Functioning Capacities",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <CategorySliders
              databaseNamesArray={
                Array [
                  "exe_1_orient",
                  "exe_1_inhibition",
                ]
              }
              slidersArray={
                Array [
                  "Orient EYES-EARS-SELF to GOAL",
                  "Inhibition",
                ]
              }
            />,
            <CategorySliders
              databaseNamesArray={
                Array [
                  "exe_2a_working",
                  "exe_2a_shifting",
                ]
              }
              slidersArray={
                Array [
                  "Working Memory",
                  "Shifting",
                ]
              }
            />,
            <CategorySliders
              databaseNamesArray={
                Array [
                  "exe_2b_verbal",
                  "exe_2b_spatial",
                  "exe_2b_general",
                  "exe_2b_global",
                  "exe_2b_flexibility",
                ]
              }
              slidersArray={
                Array [
                  "Verbal",
                  "Spatial",
                  "General",
                  "Global/Local",
                  "Flexibility",
                ]
              }
            />,
            <CategorySliders
              databaseNamesArray={
                Array [
                  "exe_3_initiation",
                  "exe_3_planning",
                  "exe_3_sequencing",
                  "exe_3_monitoring",
                ]
              }
              slidersArray={
                Array [
                  "Initiation",
                  "Planning",
                  "Sequencing",
                  "Monitoring",
                ]
              }
            />,
            <CategorySliders
              databaseNamesArray={
                Array [
                  "exe_4a_future",
                  "exe_4a_spatial",
                  "exe_4a_problem",
                  "exe_4a_reflective",
                  "exe_4a_internalized",
                ]
              }
              slidersArray={
                Array [
                  "Future-Oriented/Goal-Directed",
                  "Spatial/Temporal Organization",
                  "Problem Solving",
                  "Reflective Thinking",
                  "Internalized Language",
                ]
              }
            />,
            <CategorySliders
              databaseNamesArray={
                Array [
                  "exe_4b_motivational",
                  "exe_4b_effortful",
                  "exe_4b_self_control",
                  "exe_4b_self_regulation",
                  "exe_4b_self_compassion",
                ]
              }
              slidersArray={
                Array [
                  "Motivational Bias",
                  "Effortful Control",
                  "Self-Control",
                  "Self-Regulation",
                  "Self-Compassion",
                ]
              }
            />,
          ],
          "className": "category-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "databaseNamesArray": Array [
                "exe_1_orient",
                "exe_1_inhibition",
              ],
              "slidersArray": Array [
                "Orient EYES-EARS-SELF to GOAL",
                "Inhibition",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "databaseNamesArray": Array [
                "exe_2a_working",
                "exe_2a_shifting",
              ],
              "slidersArray": Array [
                "Working Memory",
                "Shifting",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "databaseNamesArray": Array [
                "exe_2b_verbal",
                "exe_2b_spatial",
                "exe_2b_general",
                "exe_2b_global",
                "exe_2b_flexibility",
              ],
              "slidersArray": Array [
                "Verbal",
                "Spatial",
                "General",
                "Global/Local",
                "Flexibility",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "databaseNamesArray": Array [
                "exe_3_initiation",
                "exe_3_planning",
                "exe_3_sequencing",
                "exe_3_monitoring",
              ],
              "slidersArray": Array [
                "Initiation",
                "Planning",
                "Sequencing",
                "Monitoring",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "databaseNamesArray": Array [
                "exe_4a_future",
                "exe_4a_spatial",
                "exe_4a_problem",
                "exe_4a_reflective",
                "exe_4a_internalized",
              ],
              "slidersArray": Array [
                "Future-Oriented/Goal-Directed",
                "Spatial/Temporal Organization",
                "Problem Solving",
                "Reflective Thinking",
                "Internalized Language",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "databaseNamesArray": Array [
                "exe_4b_motivational",
                "exe_4b_effortful",
                "exe_4b_self_control",
                "exe_4b_self_regulation",
                "exe_4b_self_compassion",
              ],
              "slidersArray": Array [
                "Motivational Bias",
                "Effortful Control",
                "Self-Control",
                "Self-Regulation",
                "Self-Compassion",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SessionHeader
            headings={Array []}
            title="Executive Functioning Capacities"
          />,
          <div
            className="category-content"
          >
            <CategorySliders
              databaseNamesArray={
                Array [
                  "exe_1_orient",
                  "exe_1_inhibition",
                ]
              }
              slidersArray={
                Array [
                  "Orient EYES-EARS-SELF to GOAL",
                  "Inhibition",
                ]
              }
            />
            <CategorySliders
              databaseNamesArray={
                Array [
                  "exe_2a_working",
                  "exe_2a_shifting",
                ]
              }
              slidersArray={
                Array [
                  "Working Memory",
                  "Shifting",
                ]
              }
            />
            <CategorySliders
              databaseNamesArray={
                Array [
                  "exe_2b_verbal",
                  "exe_2b_spatial",
                  "exe_2b_general",
                  "exe_2b_global",
                  "exe_2b_flexibility",
                ]
              }
              slidersArray={
                Array [
                  "Verbal",
                  "Spatial",
                  "General",
                  "Global/Local",
                  "Flexibility",
                ]
              }
            />
            <CategorySliders
              databaseNamesArray={
                Array [
                  "exe_3_initiation",
                  "exe_3_planning",
                  "exe_3_sequencing",
                  "exe_3_monitoring",
                ]
              }
              slidersArray={
                Array [
                  "Initiation",
                  "Planning",
                  "Sequencing",
                  "Monitoring",
                ]
              }
            />
            <CategorySliders
              databaseNamesArray={
                Array [
                  "exe_4a_future",
                  "exe_4a_spatial",
                  "exe_4a_problem",
                  "exe_4a_reflective",
                  "exe_4a_internalized",
                ]
              }
              slidersArray={
                Array [
                  "Future-Oriented/Goal-Directed",
                  "Spatial/Temporal Organization",
                  "Problem Solving",
                  "Reflective Thinking",
                  "Internalized Language",
                ]
              }
            />
            <CategorySliders
              databaseNamesArray={
                Array [
                  "exe_4b_motivational",
                  "exe_4b_effortful",
                  "exe_4b_self_control",
                  "exe_4b_self_regulation",
                  "exe_4b_self_compassion",
                ]
              }
              slidersArray={
                Array [
                  "Motivational Bias",
                  "Effortful Control",
                  "Self-Control",
                  "Self-Regulation",
                  "Self-Compassion",
                ]
              }
            />
          </div>,
        ],
        "className": "Executive",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "headings": Array [],
            "title": "Executive Functioning Capacities",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <CategorySliders
                databaseNamesArray={
                  Array [
                    "exe_1_orient",
                    "exe_1_inhibition",
                  ]
                }
                slidersArray={
                  Array [
                    "Orient EYES-EARS-SELF to GOAL",
                    "Inhibition",
                  ]
                }
              />,
              <CategorySliders
                databaseNamesArray={
                  Array [
                    "exe_2a_working",
                    "exe_2a_shifting",
                  ]
                }
                slidersArray={
                  Array [
                    "Working Memory",
                    "Shifting",
                  ]
                }
              />,
              <CategorySliders
                databaseNamesArray={
                  Array [
                    "exe_2b_verbal",
                    "exe_2b_spatial",
                    "exe_2b_general",
                    "exe_2b_global",
                    "exe_2b_flexibility",
                  ]
                }
                slidersArray={
                  Array [
                    "Verbal",
                    "Spatial",
                    "General",
                    "Global/Local",
                    "Flexibility",
                  ]
                }
              />,
              <CategorySliders
                databaseNamesArray={
                  Array [
                    "exe_3_initiation",
                    "exe_3_planning",
                    "exe_3_sequencing",
                    "exe_3_monitoring",
                  ]
                }
                slidersArray={
                  Array [
                    "Initiation",
                    "Planning",
                    "Sequencing",
                    "Monitoring",
                  ]
                }
              />,
              <CategorySliders
                databaseNamesArray={
                  Array [
                    "exe_4a_future",
                    "exe_4a_spatial",
                    "exe_4a_problem",
                    "exe_4a_reflective",
                    "exe_4a_internalized",
                  ]
                }
                slidersArray={
                  Array [
                    "Future-Oriented/Goal-Directed",
                    "Spatial/Temporal Organization",
                    "Problem Solving",
                    "Reflective Thinking",
                    "Internalized Language",
                  ]
                }
              />,
              <CategorySliders
                databaseNamesArray={
                  Array [
                    "exe_4b_motivational",
                    "exe_4b_effortful",
                    "exe_4b_self_control",
                    "exe_4b_self_regulation",
                    "exe_4b_self_compassion",
                  ]
                }
                slidersArray={
                  Array [
                    "Motivational Bias",
                    "Effortful Control",
                    "Self-Control",
                    "Self-Regulation",
                    "Self-Compassion",
                  ]
                }
              />,
            ],
            "className": "category-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "databaseNamesArray": Array [
                  "exe_1_orient",
                  "exe_1_inhibition",
                ],
                "slidersArray": Array [
                  "Orient EYES-EARS-SELF to GOAL",
                  "Inhibition",
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "databaseNamesArray": Array [
                  "exe_2a_working",
                  "exe_2a_shifting",
                ],
                "slidersArray": Array [
                  "Working Memory",
                  "Shifting",
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "databaseNamesArray": Array [
                  "exe_2b_verbal",
                  "exe_2b_spatial",
                  "exe_2b_general",
                  "exe_2b_global",
                  "exe_2b_flexibility",
                ],
                "slidersArray": Array [
                  "Verbal",
                  "Spatial",
                  "General",
                  "Global/Local",
                  "Flexibility",
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "databaseNamesArray": Array [
                  "exe_3_initiation",
                  "exe_3_planning",
                  "exe_3_sequencing",
                  "exe_3_monitoring",
                ],
                "slidersArray": Array [
                  "Initiation",
                  "Planning",
                  "Sequencing",
                  "Monitoring",
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "databaseNamesArray": Array [
                  "exe_4a_future",
                  "exe_4a_spatial",
                  "exe_4a_problem",
                  "exe_4a_reflective",
                  "exe_4a_internalized",
                ],
                "slidersArray": Array [
                  "Future-Oriented/Goal-Directed",
                  "Spatial/Temporal Organization",
                  "Problem Solving",
                  "Reflective Thinking",
                  "Internalized Language",
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "databaseNamesArray": Array [
                  "exe_4b_motivational",
                  "exe_4b_effortful",
                  "exe_4b_self_control",
                  "exe_4b_self_regulation",
                  "exe_4b_self_compassion",
                ],
                "slidersArray": Array [
                  "Motivational Bias",
                  "Effortful Control",
                  "Self-Control",
                  "Self-Regulation",
                  "Self-Compassion",
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
